File: 0001_0100/0039/0039.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    p = [int(x) for x in f.read().split()[1:]]
    s = 0
    while p:
        m = p.index(max(p))
        s += (m + 1) * p[m]
        p = p[m + 1:]
    q.write(str(s))

File: 0001_0100/0039/0039.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int
	fmt.Fscan(f, &n)
	p := make([]int, n)
	for i := range p {
		fmt.Fscan(f, &p[i])
	}
	s := 0
	for i := 0; i < n; i++ {
		m := i
		for j := i; j < n; j++ {
			if p[j] > p[m] {
				m = j
			}
		}
		s += (m - i + 1) * p[m]
		i = m
	}
	fmt.Fprint(q, s)
}
